lib/matplotlib/axes/_axes.py

```diff
@@ -4263,18 +4260,17 @@ def _parse_scatter_color_args(c, edgecolors, kwargs, xshape, yshape,
             try:  # Then is 'c' acceptable as PathCollection facecolors?
                 colors = mcolors.to_rgba_array(c)
                 n_elem = colors.shape[0]
-                if colors.shape[0] not in (0, 1, xsize, ysize):
+                if colors.shape[0] not in (0, 1, xsize):
                     # NB: remember that a single color is also acceptable.
                     # Besides *colors* will be an empty array if c == 'none'.
                     valid_shape = False
                     raise ValueError
-            except ValueError:
+            except (ValueError, TypeError):
```

Comment:
where does the typeerror come from?

Reply:
```python
In [9]: mcolors.to_rgba_array(None)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-9-1ff544d60239> in <module>
----> 1 mcolors.to_rgba_array(None)

~/work/programs/py/mpl/matplotlib/lib/matplotlib/colors.py in to_rgba_array(c, alpha)
    282         pass
    283     # Convert one at a time.
--> 284     result = np.empty((len(c), 4), float)
    285     for i, cc in enumerate(c):
    286         result[i] = to_rgba(cc, alpha)

TypeError: object of type 'NoneType' has no len()
```

Reply:
Not to be a pain... but can `c` really be None at that point?  If so, why didn't the previous version need to handle that, or what example (what scatter() call) would cause an uncaught exception in the previous version?

Reply:
It's a reasonable question.  I added the TypeError because a test was otherwise failing, if I remember correctly.  (I can't imagine any other reason I would have put it in.)  I think it was because of a None.  I will have to take out the TypeError and re-run the tests to find out.

Reply:
Trying it now without catching the TypeError, tests pass on my machine, so maybe it was needed only at an intermediate stage while I was working on the PR.  I've reverted that part of the change.  Let's see if the tests still pass here, apart from the unrelated docs problem.